rest/class-boldgrid-backup-rest-archive.php

```diff
@@ -0,0 +1,277 @@
+					'type'        => 'string',
+				],
+				'title'         => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Name of the archive.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'description'   => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Description of the archive.', 'boldgrid-backup' ),
+					'type'        => 'array',
+				],
+				'filename'      => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Filename of backup.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'url'           => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Download link for the backup.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+				'creation_date' => [
+					'context'     => [ 'view' ],
+					'description' => esc_html__( 'Date the archive was created.', 'boldgrid-backup' ),
+					'type'        => 'string',
+				],
+			],
+		];
+
+		return $schema;
+	}
+
+	/**
+	 * Prepare the item for the REST response.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param mixed $item WordPress representation of the item.
+	 * @param WP_REST_Request $request Request object.
+	 * @return mixed
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		return $this->filter_schema_properties( $item );
+	}
+
+	/**
+	 * Create a new archive.
+	 *
+	 * For examples, please see rest/README.MD
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see Boldgrid_Backup_Rest_Controller::permission_check for more information if you are getting
+	 *                                                        a 401 unauthorized error.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return array                   Job Resource.
+	 */
+	public function create_item( $request ) {
+		// Initialize a new task.
+		$task = new Boldgrid_Backup_Admin_Task();
+		$task->init( [ 'type' => 'backup' ] );
+		$task->update();
+		$task->date_format = 'c';
+
+		// Trigger our backup.
+		$nopriv = new Boldgrid_Backup_Admin_Nopriv();
+		$nopriv->do_backup( [ 'task_id' => $task->get_id() ] );
+
+		return new WP_REST_Response( $task->get(), 200 );
+	}
+
+	/**
+	 * Get all archives for a WordPress.
+	 *
+	 * For examples, please see rest/README.MD
+	 *
+	 * @since X.X.X
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return array                   A collection of archive resources.
+	 */
+	public function get_items( $request ) {
+		// Init our list of raw backup data.
+		$core = apply_filters( 'boldgrid_backup_get_core', null );
+		$core->archives_all->init();
+
+		// Init our list of backups to return.
+		$backups = [];
+
+		foreach ( $core->archives_all->all as $backup ) {
+			$archive = Boldgrid\Backup\Archive\Factory::get_by_filename( $backup['filename'] );
+			$url     = $archive->generate_download_link( $backup['filename'] );
+
+			$backups[] = [
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.